### PR TITLE
Update to Homie Convention v2.0.1

### PR DESCRIPTION
**Pull request by timpur**

### CODE REVIEW

File: src/Homie/Boot/BootConfig.cpp [hunk 1]
```diff
@@ -335,7 +335,7 @@ void BootConfig::_onDeviceInfoRequest(AsyncWebServerRequest *request) {
   for (IHomieSetting* iSetting : IHomieSetting::settings) {
     JsonObject& jsonSetting = jsonBuffer.createObject();
 
-    if (String(iSetting->getType()) != "unknown") {
+    if (strcmp(iSetting->getType(), "unknown") != 0) {
```
Comment (euphi) [inline src/Homie/Boot/BootConfig.cpp]: Why strcmp?

Comment (timpur) [reply src/Homie/Boot/BootConfig.cpp]: In the core of the lib, we should try avoid using String. String will copy the chars into dynamic memory even for something as simple as this... Using String everywhere can lead to heep fragmentation.

**Comment by bodiroga** [comment]
Hi @timpur!

Do you plan to include this PR in the final Homie-ESP8266 V2.0 version?

Thanks!

**Comment by timpur** [comment]
Not sure, what do you guy think ?

**Comment by euphi** [comment]
Yes, of course. It is ready now, so it should be added.

**Pull request merged**